### ISSUE TITLE
http and admin are included but never started

### ISSUE THREAD

**Issue by prime31** [SanderMertens/ecs_collisions#2]
Sorry about the lack of PR for this one. I'm not on a computer where I can fork/fix/push.

I believe this line https://github.com/SanderMertens/ecs_collisions/blob/master/src/main.c#L25 was meant to be `ecs_world_t *world = ecs_init_w_args(argc, argv);` so that the admin interface can be used (at least that is what I am assuming because civetweb and admin are both included).

On an related side note, I am not sure if this is intended or not but the demo only renders the spinning rectangle for me. The circles and squares created are not rendered. I'm just exploring flecs and havent written any C code in ages so I can't seem to make out where the issue is.

<img width="799" alt="Screen Shot 2019-04-30 at 12 51 33 PM" src="https://user-images.githubusercontent.com/60377/56989231-d069fb00-6b46-11e9-9aec-364163b2ff47.png">

**Comment by SanderMertens** [comment]
This looks like the issue I'm looking into at the moment that was introduced as a result of https://github.com/SanderMertens/flecs/commit/fba802d1757d1a4d78020aa8486fd13e7d013f2f

I'll post an update here when I fixed the issue

**Comment by SanderMertens** [comment]
You are correct btw, to start the admin from the command line the `ecs_init_w_args` should be used. I'll update the example with it.

**Comment by SanderMertens** [comment]
I just fixed the issue. If you use the latest version of flecs the example should work again.

**Comment by prime31** [comment]
That was quick! Works like a champ now. Thanks!

**Comment by prime31** [comment]
Quick question for you, feel free to answer it with "Google it you idiot" if that is the correct answer ;) It's been a long time since I worked with C and that was all via Xcode from pre-generated projects that "just worked" with Xcode's debugger.

How do you debug a Flecs app that is built with Bake with an IDE and get autocomplete working with the project? I am using Visual Studio Code and cant seem to figure out how to get it to parse and debug a Flecs/Bake project correctly so any tips/links to point me in the right direction would be great. Or just tell me to Google it again ;)

**Comment by SanderMertens** [comment]
What you can try is, open visual studio settings, and search for "includePath". This should bring up the configuration. Confusingly, if you click on "Edit in settings.json" it won't actually add the include path, you need to do that manually.

If I remember correctly, you should add something along the lines of:
```
"C_Cpp.default.includePath": ["<path>"]
```

The include path you'll want to include is `~/bake/include` (replace `~` with your home path- not sure if vscode is smart enough to parse `~`). This contains all the include files for your projects. I've had to play around a bit with it to get it to work, and it doesn't always correctly resolve functions, but for most of the time it works reasonably well.

**Comment by prime31** [comment]
So, I'm having a rough time trying to get an environment setup with proper code completion and debugging. I've tried CLion, VS Code and Atom with no luck. What exactly is the setup you are using to build and debug Flecs apps? Maybe if I mimic it I might have better luck...

**Comment by SanderMertens** [comment]
I had to experiment a bit, but I got it to work with the latest version of vscode.

First make sure you have the C/C++ IntelliSense extension installed. Then go to "Code" (the menu) -> Preferences -> Settings. In the window that opens, click on the `{ }` to modify the JSON directly.

My exact configuration here is:

```
{
    "git.enableSmartCommit": true,
    "window.zoomLevel": 0,
    "editor.rulers": [
        80
    ],
    "explorer.confirmDragAndDrop": false,
    "C_Cpp.default.cStandard": "c99",
    "C_Cpp.default.defines": ["__BAKE__"],
    "C_Cpp.default.includePath": [
        "/Users/sandermertens/bake/include",
        "${workspaceFolder}"
    ],
    "C_Cpp.updateChannel": "Insiders"
}
```
This should work for any bake project. The `${workspaceFolder}` makes sure the editor can resolve the `<include/project.h>`, and all other include files should be accessible from the bake path. The `__BAKE__` define will make sure that vscode tries to include the generated include statements in `bake_config.h`.

Let me know if this works! Also, consider joining the gitter channel (https://gitter.im/flecsdev/community) if you have more questions :)
